ocfweb/context_processors.py

```diff
@@ -3,13 +3,18 @@
 
 from ocflib.lab.hours import DayHours
 
+from ocfweb.component.banner import get_banner_message
+
 
 def ocf_template_processor(request):
     now = datetime.now()
     today = date.today()
     hours = DayHours.from_date(today)
 
+    banner = get_banner_message()
+
     return {
         'lab_is_open': hours.is_open(now),
         'current_lab_hours': hours,
+        'force_lab_closed': banner.visible and banner.force_lab_closed,
```

Comment:
OK, but they're still logically tied by being in the same file, which is
fine. You also have to update both fields when restoring to normal, which
is also fine. Maybe this will evolve into a more general thing.
On Sep 26, 2015 12:41 AM, "Chris Kuehl" <notifications@github.com> wrote:

> In ocfweb/context_processors.py
> <https://github.com/ocf/ocfweb/pull/34#discussion_r40491223>:
>
> >      return {
> >          'lab_is_open': hours.is_open(now),
> >          'current_lab_hours': hours,
> > +        'force_lab_closed': banner.visible and banner.force_lab_closed,
>
> I think all we'd need is
> force_lab_closed: true/false
>
> You can just set it to false when the lab shouldn't be forced closed.
>
> You're right, you can have the banner but not close the lab. But I don't
> think you can close the lab without a banner? (Not that we probably need
> to, but I think it makes sense to be able to)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ocf/ocfweb/pull/34/files#r40491223>.
>